webui/react/src/pages/ExperimentList.tsx

```diff
@@ -564,23 +571,55 @@ const ExperimentList: React.FC = () => {
     resetSettings([ ...filterKeys, 'tableOffset' ]);
   }, [ resetSettings ]);
 
-  const handleUpdateColumns = useCallback((columns: string[]) => {
-    updateSettings({ columns: columns.length === 0 ? [ 'name' ] : columns });
+  const handleUpdateColumns = useCallback((columns: ExperimentColumnName[]) => {
+    if (columns.length === 0) {
+      updateSettings({
+        columns: [ 'id', 'name' ],
+        columnWidths: [
+          DEFAULT_COLUMN_WIDTHS['id'],
+          DEFAULT_COLUMN_WIDTHS['name'],
+        ],
+      });
+    } else {
+      updateSettings({
+        columns: columns,
+        columnWidths: columns.map((col) => DEFAULT_COLUMN_WIDTHS[col]),
+      });
+    }
   }, [ updateSettings ]);
 
   const { modalOpen } = useModalCustomizeColumns({
     columns: transferColumns,
     defaultVisibleColumns: DEFAULT_COLUMNS,
-    onSave: handleUpdateColumns,
+    onSave: (handleUpdateColumns as (columns: string[]) => void),
   });
 
+  const resetColumnWidths = useCallback(
+    () =>
+      updateSettings({ columnWidths: settings.columns.map((col) => DEFAULT_COLUMN_WIDTHS[col]) }),
+    [ settings.columns, updateSettings ],
+  );
+
   const openModal = useCallback(() => {
     modalOpen({ initialVisibleColumns: settings.columns });
   }, [ settings.columns, modalOpen ]);
 
   const switchShowArchived = useCallback((showArchived: boolean) => {
-    updateSettings({ archived: showArchived, row: undefined });
-  }, [ updateSettings ]);
+    if (!showArchived || settings.columns?.includes('archived')) {
+      updateSettings({ archived: showArchived, row: undefined });
+    } else {
+      if (!settings.columns?.length || !settings.columnWidths?.length) return;
+      const columns = [ ...settings.columns, 'archived' ] as ExperimentColumnName[];
+      const columnWidths = [ ...settings.columnWidths, DEFAULT_COLUMN_WIDTHS['archived'] ];
+      updateSettings({
+        archived: showArchived,
+        columns: columns,
+        columnWidths: columnWidths,
+        row: undefined,
+      });
+    }
```

Comment:
Oh, this is probably where the archived toggle is breaking. Not 100% sure which line of code is causing it, but it's definitely broken.

Reply:
I have restored the original behavior of archived toggle/column